tests/unit/closeable_test.cc

```diff
@@ -246,6 +246,19 @@ SEASTAR_TEST_CASE(with_stoppable_exception_test) {
     });
 }
 
+SEASTAR_THREAD_TEST_CASE(move_gate_test) {
+    gate g1;
+    g1.enter();
+    gate g2 = std::move(g1);
+    // the state in g1 should be moved into g2
+    BOOST_CHECK_EQUAL(g1.get_count(), 0);
+    BOOST_REQUIRE_EQUAL(g2.get_count(), 1);
+    g2.leave();
+    g2.close().get();
+    BOOST_CHECK(!g1.is_closed());
+    BOOST_CHECK(g2.is_closed());
+}
```

Comment:
This is a test case for moving open gate. I think we also need a test for moving closing and closed gates.

Reply:
added a test for both of these cases.